### ISSUE TITLE
Errors using BLAKE2 and hkdf.

### ISSUE THREAD

**Issue by ronaldmickey** [weidai11/cryptopp#810]
I seem to have a function (below) that compiles fine but crashes when ran

    #include <iostream>
    #include <string>
    #include <cryptopp/cryptlib.h>
    #include <cryptopp/hkdf.h>
    #include <cryptopp/blake2.h>
    #include <cryptopp/filters.h>
	std::string hkdf(std::string pass, std::string seed)
	{
		std::string decoded1;
		std::string decoded2;

		CryptoPP::StringSource ss1(pass, true /*pumpAll*/,
			new CryptoPP::StringSink(decoded1));
		const CryptoPP::byte* password = reinterpret_cast<const CryptoPP::byte*>(decoded1.data());

		CryptoPP::StringSource ss2(seed, true /*pumpAll*/,
			new CryptoPP::StringSink(decoded2));
		const CryptoPP::byte* salt = reinterpret_cast<const CryptoPP::byte*>(decoded2.data());

		size_t slen = strlen((const char*)salt);
		size_t plen = strlen((const char*)password);

		CryptoPP::byte info[] = "HKDF key derivation";
		size_t ilen = strlen((const char*)ilen);

		CryptoPP::byte derived[CryptoPP::BLAKE2b::DIGESTSIZE];
		CryptoPP::HKDF<CryptoPP::BLAKE2b> hkdf; // CryptoPP::HKDF<CryptoPP::BLAKE2s> hkdf; fails also,
		hkdf.DeriveKey(derived, sizeof(derived), password, plen, salt, slen, info,
			ilen);

		return std::string((char*)derived, CryptoPP::BLAKE2b::DIGESTSIZE);
	}

**Comment by noloader** [comment]
I believe `size_t ilen = strlen((const char*)ilen);` is undefined behavior.

**Comment by ronaldmickey** [comment]
https://www.cryptopp.com/wiki/HKDF its basically a copy and paste of the sample here.

**Comment by ronaldmickey** [comment]
I assume `size_t ilen = strlen((const char*)info);` would be what was intended?

Correcting that still doesn't fix the issue at hand.

The exact error i'm getting is `Unhandled exception at 0x00007FF86156A388 in app.exe: Microsoft C++ exception: CryptoPP::InvalidArgument at memory location 0x0000003FD28FF370.`

I've tried other hash types and all but the BLAKE2b and BLAKE2s work. Specifcally I've tried:
'tiger'
'sha1'
'sha_224'
'sha_256'
'sha_384'
'sha_512'
'sha3_224'
'sha3_256'
'sha3_384'
'sha3_512'
'keccak_224'
'keccak_256'
'keccak_384'
'keccak_512'
'blake2b'
'blake2s'
'sm3'
'whirlpool'
'ripemd_128'
'ripemd_160'
'ripemd_256'
'ripemd_320'
